File: ground-sensor/groundsensor.c
#include <stdio.h>
#include <stdint.h> 
#include <sys/types.h>
#include <sys/stat.h>
#include <fcntl.h>
#include <linux/i2c-dev.h> /* for I2C_SLAVE */
//#include <linux/i2c.h>
#include <sys/ioctl.h>
#include <stdlib.h>
#include <unistd.h>

int fh;
uint8_t i = 0;
int result[6];

int read_reg(int file, uint8_t reg, int count) {
	int8_t data[2];

	if (count < 1 || count > 2)
		return -1;

	data[0] = reg;

	if (write(file, &data, 1) != 1) {
		perror("write before read");
		return -1;
	}

	data[1] = 0;

	if (read(file, &data, count) != count) {
		perror("read");
		return -1;
	}

	return (data[1] << 8) + data[0];
}

int main() {

	fh = open("/dev/i2c-4", O_RDWR);	// open the I2C dev driver for bus 3

	ioctl(fh, I2C_SLAVE, 0x60);			// tell the driver we want the device with address 0x60 on the I2C bus

	while(1) {

		for(i=0; i<6; i++) {				// read the first 6 registers => 3 sensors value
			result[i] = read_reg(fh, i, 1);	// read 1 byte register
		}

		printf("%d,%d,%d\n", result[0]*256+result[1], result[2]*256+result[3], result[4]*256+result[5]);
		
		//if(abs(result[0]*256+result[1]-700)>80 || abs(result[2]*256+result[3]-700)>80 || abs(result[4]*256+result[5]-700)>80) {
		//	printf("comm error: %d,%d,%d\n", result[0]*256+result[1], result[2]*256+result[3], result[4]*256+result[5]);
		//}

		usleep(20000);	// let the bus be free for a while

	}

	close(fh);
	
	return 0;

}

File: ground-sensor/groundsensor.py
from smbus2 import SMBus, i2c_msg
import sys
import time

I2C_CHANNEL = 12
LEGACY_I2C_CHANNEL = 4
GROUNDSENSOR_ADDRESS = 0x60  # Device address

groundData = bytearray([0] * 6)
groundValue = [0 for x in range(3)]
bus = None
	
def read_reg(reg, count):
	global bus
	global imu_addr
	data = None
	
	try:			
		data = bus.read_i2c_block_data(GROUNDSENSOR_ADDRESS, reg, count)
		#print(data)
	except:
		print("read error")
		return None

	return data;

try:
	bus = SMBus(I2C_CHANNEL)
except:
	try:
		bus = SMBus(LEGACY_I2C_CHANNEL)
	except:
		print("Cannot open I2C device")
		sys.exit(1)

while 1:

	start = time.time()

	groundData = read_reg(0, 6)

	groundValue[0] = (groundData[1] + (groundData[0]<<8))
	groundValue[1] = (groundData[3] + (groundData[2]<<8))
	groundValue[2] = (groundData[5] + (groundData[4]<<8))

	print("left={0:>3d}, center={1:>3d}, right={2:>3d}".format(groundValue[0], groundValue[1], groundValue[2]))

	# Communication frequency @ 20 Hz.
	time_diff = time.time() - start
	if time_diff < 0.050:
		time.sleep(0.050 - time_diff);
